omnibus/config/software/msodbcsql18.rb

```diff
@@ -10,26 +10,39 @@
 
 # Dynamically build source url and sha256 based on the linux platform and architecture
 # The source url and sha256 are taken from the official Microsoft ODBC Driver for SQL Server page:
+source_url_base = "https://packages.microsoft.com"
+
 if debian_target?
+  base_package_name = "#{name}_#{version}"
   if arm_target?
     # Debian ARM64 build
-    package_name = "#{name}_#{version}_arm64.deb"
-    source url: "https://packages.microsoft.com/debian/12/prod/pool/main/m/msodbcsql18/#{package_name}"
-    source sha256: "d9bb2d2e165e9d86f6b5de75b2baa24c9f0f25107471bcf82254d27f5dafff30"
+    package_name = "#{base_package_name}_arm64.deb"
+    source sha256: "37e692b1517f1229042c743d0f2a7191e0dcb956bbc3785a895aaa6dc328467e"
+    source url: "#{source_url_base}/debian/12/prod/pool/main/m/msodbcsql18/#{package_name}"
   else
     # Debian AMD64 build
-    package_name = "#{name}_#{version}_amd64.deb"
-    source url: "https://packages.microsoft.com/debian/12/prod/pool/main/m/msodbcsql18/#{package_name}"
-    source sha256: "ae8eea58236e46c3f4eae05823cf7f0531ac58f12d90bc24245830b847c052ee"
+    package_name = "#{base_package_name}_amd64.deb"
+    source sha256: "f91004ce72fcd92e686154f90e2a80f4f86469e7cb5f42ef79cba79dc6727890"
+    source url: "#{source_url_base}/debian/12/prod/pool/main/m/msodbcsql18/#{package_name}"
```

Comment:
I wonder it things shouldn't be cleaned up in order not to have to download a different package based on the distributiond. I'd expect the driver itself to be identical, meaning you could always download the same package regardless of the target.

Reply:
MS published package with different name, i.e. https://packages.microsoft.com/debian/11/prod/pool/main/m/msodbcsql18/msodbcsql18_18.3.2.1-1_amd64.deb vs. https://packages.microsoft.com/debian/11/prod/pool/main/m/msodbcsql18/msodbcsql18_18.3.3.1-1_amd64.deb. Since we couldn't use `apt install` to download and install the package, the package name needs to be specified in the download url.

Reply:
Sorry I was probably unclear, what I meant is you could download & extract a single package, deb or rpm regardless of the current target, and extract the driver from it, as opposed to downloading the debian package when running on debian, and downloading the rpm package when running on centos/RHEL

Reply:
Also to clarify, this doesn't have to happen on this PR, but it will make your life easier during next updates

Reply:
I manually extracted the driver from a debian package and put it on redhat. But `unixODBC` was not happy with it. I wasn't quite sure what's really breaking. I will try the idea of extract from single package in a separate PR.